libs/shared/util-loaders/src/context/LoadContext.ts

```diff
@@ -12,3 +12,13 @@ export type LoadContext = {
      */
     limit: number;
 };
+
+/**
+ * Broadens the definition of Load context to allow for the temporary use of
+ * number cursors. This type should only be used inside of a loader class
+ * as a number cursor will cause downstream problems if Prisma get's
+ * ahold of it
+ */
+export interface InternalLoadContext extends Omit<LoadContext,"cursor"> {
```

Comment:
Can you explain this a bit? Cursors are intentionally opaque `string`s.

Reply:
Yeah totally. The cursor's type in the gql query is a number so it was failing unless I converted it into a number before putting it into the query (it also failed when I tried to tell the gql function to use a string). There might be a better way to do this, but this is what worked for me. I made the interface because the obvious place to change it to a number was in the preLoad function, and so I needed to return it from there (with the correct typing) before it was used in the load function.

Reply:
I see. The problem with this is that it adds unnecessary types to the mix and it also breaks the abstraction. A `cursor` is an opaque string and it has to stay that way. For example, on the API we can create a composite cursor (one that has 2 values in it) because we treat all cursors as opaque strings. This is something that you just can't do with `number`s. If we start adding specialized types we'll end up with a brittle inheritance hierarchy sooner or later and a lot of corner cases.

This is the code for `loadRaw` that will execute the actual loading:

```ts
protected loadRaw(context: LoadContext) {
    return this.client.query(this.graphQLQuery, context, this.codec);
}
```

The whole loader abstraction is just an implementation of the [Template Method](https://en.wikipedia.org/wiki/Template_method_pattern) pattern so what we can do is to put a new template in there that converts the `context` into a shape that's accepted by the `query` function. `query` takes a `Record<string, any>` so any JSON object will do. We can call this `extractQueryParams` for example and the default implementation can just pass the `context` as-is (like we do in `loadRaw`).

What do you think?

Reply:
That makes sense to me! I'll go ahead and change it to use this method

Reply:
Done